### PR TITLE
[#1007] Connection through EDI and training on Tekton

### PR DESCRIPTION
**Pull request by mcdoker18**
* Added HTTP golang repositories for connections, trainings
  and packagings
* Collect logging from all containers(setup,main,result)
* Fixed unstable unit and robot tests
* Packaging improvements:
  * Moved extraction of all packages entities from operator to setup CLI
* Training improvements:
  * Switched training process on tekton
  * Got rid of saving ssh keys in k8s secret
  * Moved extraction of all training entities from operator to setup CLI
* Others:
  * Updated swag to the 1.6.3 version
  * Removed buildah from the packager and trainer Docker images and
    golang codebase due to switching on tekton

### CODE REVIEW

File: legion/operator/pkg/controller/modeltraining/tectoncd.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	toolchainIntegration *training.ToolchainIntegration,
+) (*tektonv1alpha1.TaskSpec, error) {
+	mainPackagerStep, err := createMainPackagerStep(trainingCR, toolchainIntegration)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tektonv1alpha1.TaskSpec{
+		Steps: []tektonv1alpha1.Step{
+			createInitPackagerStep(trainingCR.Name),
```
Comment (vlad-tokarev) [inline legion/operator/pkg/controller/modeltraining/tectoncd.go]: Maybe we should rename these func names to blabla**Trainer**Step ?

Comment (mcdoker18) [reply legion/operator/pkg/controller/modeltraining/tectoncd.go]: Fixed

File: legion/operator/pkg/rclone/type.go [hunk 2]
```diff
@@ -139,13 +139,19 @@ func (os *ObjectStorage) Download(localPath, remotePath string) error {
...
+			"local_dir", localDir, "remote_dir", remotePath,
+		)
 		return sync.CopyDir(localFs, remoteFs, copyEmptySrcDirs)
 	}
 
 	if len(localFileName) == 0 {
 		localFileName = srcFileName
 	}
 
+	log.Info("Download the file file from the remote bucket",
```
Comment (maxkochubey) [inline legion/operator/pkg/rclone/type.go]: "file file"

File: legion/operator/pkg/rclone/type.go [hunk 3]
```diff
@@ -186,12 +192,18 @@ func (os *ObjectStorage) Upload(localPath, remotePath string) error {
...
+			"local_dir", localPath, "remote_dir", remotePath,
+		)
 		return sync.CopyDir(rFs, lFs, copyEmptySrcDirs)
 	}
 
 	if len(remoteFileName) == 0 {
 		remoteFileName = srcFileName
 	}
 
+	log.Info("Upload the file file to the remote bucket",
```
Comment (maxkochubey) [inline legion/operator/pkg/rclone/type.go]: "file file"

**Review by karbyshevds — DISMISSED** [review]

File: legion/operator/pkg/repository/packaging/http/model_packaging.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	mpBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		mpLogger.Error(err, "Read all data from EDI response")
+
+		return nil, err
+	}
+	defer func() {
+		bodyCloseError := response.Body.Close()
+		if bodyCloseError != nil {
+			mpLogger.Error(err, "Closing model packaging response body body")
```
Comment (maxkochubey) [inline legion/operator/pkg/repository/packaging/http/model_packaging.go]: "body body"

File: legion/operator/pkg/rclone/type.go (hunk 2 above)
Comment (mcdoker18) [reply legion/operator/pkg/rclone/type.go]: fixed

File: legion/operator/pkg/repository/packaging/http/model_packaging.go (hunk 4 above)
Comment (mcdoker18) [reply legion/operator/pkg/repository/packaging/http/model_packaging.go]: fixed

File: legion/operator/pkg/rclone/type.go (hunk 3 above)
Comment (mcdoker18) [reply legion/operator/pkg/rclone/type.go]: fixed

**Review by maxkochubey — DISMISSED** [review]

**Review by maxkochubey — APPROVED** [review]

**Review by karbyshevds — APPROVED** [review]

**Pull request merged**